controllers/storagecluster/cephrbdmirrors.go

```diff
@@ -0,0 +1,158 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+
+	ocsv1 "github.com/openshift/ocs-operator/api/v1"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type ocsCephRbdMirrors struct{}
+
+func (r *StorageClusterReconciler) deleteCephRbdMirrorInstance(cephRbdMirror *cephv1.CephRBDMirror, existing *cephv1.CephRBDMirror) error {
+	var err error
+
+	if cephRbdMirror.GetDeletionTimestamp().IsZero() {
+		r.Log.Info("Deleting CephRbdMirror.", "CephRbdMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+		err = r.Client.Delete(context.TODO(), existing)
+	}
+	return err
+}
+
+// newCephRbdMirrorInstances returns the cephRbdMirror instances that should be created
+// on first run.
+func (r *StorageClusterReconciler) newCephRbdMirrorInstances(initData *ocsv1.StorageCluster) ([]*cephv1.CephRBDMirror, error) {
+
+	var rbdMirroringSpec cephv1.RBDMirroringSpec
+	rbdMirroringSpec.Count = 1
+	rbdMirroringSpec.Resources = corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("1"),
+			corev1.ResourceMemory: resource.MustParse("2Gi"),
+		},
+		Requests: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("1"),
+			corev1.ResourceMemory: resource.MustParse("2Gi"),
+		},
+	}
+
+	ret := []*cephv1.CephRBDMirror{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generateNameForCephRbdMirror(initData),
+				Namespace: initData.Namespace,
+			},
+			Spec: rbdMirroringSpec,
+		},
+	}
+	for _, obj := range ret {
+		err := controllerutil.SetControllerReference(initData, obj, r.Scheme)
+		if err != nil {
+			r.Log.Error(err, "Unable to set controller reference for CephRBDMirror.", "CephRBDMirror", klog.KRef(obj.Namespace, obj.Name))
+			return nil, err
+		}
+	}
+	return ret, nil
+}
+
+// ensureCreated ensures that cephRbdMirror resources exist in the desired
+// state.
+func (obj *ocsCephRbdMirrors) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
```

Comment:
Please refactor this function and try to keep it straightforward. It's difficult to understand what's going on.

Reply:
refactoring is done